app/src/main/java/org/openobservatory/ooniprobe/test/test/WebConnectivity.java

```diff
@@ -34,12 +34,20 @@ public WebConnectivity() {
 			measurement.is_anomaly = !json.test_keys.blocking.equals("false");
 	}
 
+	/*
+ 	if the option max_runtime is already set in the option and is not MAX_RUNTIME_DISABLED let's use it
+ 	else if the input are sets we calculate 5 seconds per input
+ 	at last we check if max_runtime is enabled, in  that case we use the value in the settings
+ 	*/
 	@Override public int getRuntime(PreferenceManager pm) {
-		if (getMax_runtime() != null)
+		if (getMax_runtime() != null  &&
+				getMax_runtime() > PreferenceManager.MAX_RUNTIME_DISABLED)
```

Comment:
What is the difference between the max runtime you get here and the one we get from the settings? Why do we have two different authorities regarding max runtime in the code base?

Reply:
In  the settings there is the user set runtime,  in the test there is the actual runtime. That can be the setting one or 5 * number_urls in case of ooni run or no max_runtime